File: app-mvp/src/main/java/com/xuie/imaginaryandroid/gui/nets/video/VideosPresenter.java
package xuk.imaginary.gui.nets.video;

import xuk.imaginary.data.source.NetsRepository;

import io.reactivex.android.schedulers.AndroidSchedulers;

import static xuk.imaginary.common.Utils.checkNotNull;

/**
 * @author xuie
 * @date 17-7-5
 */
public class VideosPresenter implements VideosContract.Presenter {
    private NetsRepository netsRepository;
    private VideosContract.View videosView;
    private int currentPage = 0;

    public VideosPresenter(NetsRepository netsRepository, VideosContract.View videosView) {
        this.netsRepository = checkNotNull(netsRepository);
        this.videosView = checkNotNull(videosView);
        this.videosView.setPresenter(this);
    }

    @Override
    public void subscribe() {
    }

    @Override
    public void unsubscribe() {

    }

    @Override
    public void getList(String type, boolean isRefresh) {
        if (isRefresh) {
            currentPage = 0;
        } else {
            currentPage += 1;
        }
        netsRepository.getVideoList(type, currentPage)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(videoBeen -> videosView.addList(isRefresh, videoBeen)
                        , Throwable::printStackTrace);

    }
}

File: app-mvp-kotlin/src/main/java/xuk/imaginary/gui/nets/video/VideosPresenter.kt
package xuk.imaginary.gui.nets.video

import io.reactivex.android.schedulers.AndroidSchedulers
import xuk.imaginary.data.source.NETSRepository

/**
 * Created by xuie on 17-7-5.
 */

class VideosPresenter(
    private val netsRepository: NETSRepository,
    private val videosView: VideosContract.View
) : VideosContract.Presenter {
  private var currentPage = 0

  init {
    this.videosView.presenter = this
  }

  override fun start() {
  }

  override fun getList(type: String, isRefresh: Boolean) {
    if (isRefresh) {
      currentPage = 0
    } else {
      currentPage += 1
    }
    netsRepository.getVideoList(type, currentPage)
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe({ videoBeen -> videosView.addList(isRefresh, videoBeen) },
            { e -> e.printStackTrace() })

  }
}
